### PR TITLE
Bump CI Xcode version to 12.5.1 and macOS to 11

### PR DESCRIPTION
**Pull request by n8chur**

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -28,8 +30,8 @@ jobs:
     steps:
```
Comment (n8chur) [inline .github/workflows/tests.yaml:30]: I can't comment on the line, but it's above on line 27/28:

Should we bump this to 14.5?

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -28,8 +30,8 @@ jobs:
...
     - uses: actions/checkout@v1
 
-    - name: Switch Xcode 12
-      run: sudo xcode-select -switch /Applications/Xcode_12.2.app
+    - name: Switch to Xcode 12.5.1
+      run: sudo xcode-select -s /Applications/Xcode_12.5.1.app
 
     - name: Install iOS ${{ matrix.sdk }}
       run: xcversion simulators --install="iOS ${{ matrix.sdk }}"
```
Comment (n8chur) [inline .github/workflows/tests.yaml:37]: I can't comment on the line, but it's above on line 27/28:

Should we bump the iOS 14 version to 14.5?

Comment (kylebshr) [reply .github/workflows/tests.yaml:37]: I think SPOS is doing that so probably should match it

Comment (kylebshr) [reply .github/workflows/tests.yaml:37]: Makes everything easier (e.g., will likely have matching sims installed locally)

**Review by kylebshr — APPROVED** [review]

**Comment by n8chur** [comment]
Re-requesting review since I bumped the `swift-version`—just double checking that that seems safe @kylebshr!

**Comment by n8chur** [comment]
I didn't bump the `Package.swift` tooling version as part of these changes but we probably want to do that soon. @watt pointed out that the 5.4 tooling change how resources are resolved so it shows a bunch of warnings we'll have to fix when we bump that.

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (n8chur) [reply .github/workflows/tests.yaml:37]: - [ ] Use 14.5 simulator

**Review by kyleve — APPROVED** [review]

**Pull request merged**